test/Semantics/select-case.f90

```diff
@@ -0,0 +1,207 @@
+  ! ***************************************************
+   select case (grade1)
+      case ('A')
+      print *, "Excellent!"
+      case ('B')
+      case ('C')
+         print *, "Well done"
+      case default
+         print *, "Invalid grade"
+   end select
+
+   select case (grade2)
+      case (1)
+      print *, "Excellent!"
+      case (2)
+      case (3)
+         print *, "Well done"
+      case default
+         print *, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case (.true.)
+       name = 'true'
+      case (.false.)
+        name = 'false'
+   end select
+   print *, "Your grade is ", name
+
+   select case (name)
+      case default
+         print *, "Invalid grade"
+      case ('now')
+      case ('test')
+         print *, "Well done"
+   end select
+  ! ***************************************************
+
+  ! C1145
+  ! ***************************************************
+  !ERROR: SELECT CASE expression must be of type CHARACTER, INTEGER, OR LOGICAL
+  select case (grade4)
+     case (1.0)
+     print *, "Excellent!"
+     case (2.0)
+     case (3.0)
+        print *, "Well done"
+     case default
+        print *, "Invalid grade"
+  end select
+
+  !ERROR: SELECT CASE expression must be of type CHARACTER, INTEGER, OR LOGICAL
+  select case (score)
+     case (score_val)
+       print *, "Half century"
+     case (scores(100))
+       print *, "Century"
+  end select
+  ! ***************************************************
+
+  ! C1146
+  ! ***************************************************
+  select case (grade3)
+     case default
+        print *, "Invalid grade"
+     case (.true.)
+        print *, "Well done"
+     !ERROR: Not more than one of the selectors of SELECT CASE statement may be DEFAULT
+     case default
+        print *, "Invalid grade"
+  end select
+  ! ***************************************************
+
+  ! C1147
+  ! ***************************************************
+  select case (grade2)
+     !ERROR: CASE value must be of type INTEGER
+     case (:'Z')
+         print *, "Excellent!"
+     case default
+         print *, "Invalid grade"
+   end select
+
+  select case (grade1)
+     !ERROR: CASE value must be of type CHARACTER
+     case (:1)
+         print *, "Excellent!"
+     case default
+         print *, "Invalid grade"
+   end select
+
+  select case (grade3)
+     case default
+        print *, "Invalid grade"
+     case (.true.)
+     !ERROR: CASE value must be of type LOGICAL
+     case (3)
+        print *, "Well done"
+  end select
+
+  select case (grade2)
+     case default
+        print *, "Invalid grade"
+     case (2 :)
+        print *, "Well done"
+     !ERROR: CASE value must be of type INTEGER
+     case (.true. :)
+  end select
+
+  select case (UTF8_var1)
+     case (UTF8_var2)
+         print *, "Excellent!"
+     !ERROR: Character kind type of case construct (=1) mismatches with the kind type of case value (=4)
+     case (UTF32_var)
```

Comment:
`CHARACTER` values are not UTF-8 in memory, and there is no such thing as UTF-16 or UTF-32.